packages/rocketchat-api/server/v1/users.js

```diff
@@ -271,6 +271,76 @@ RocketChat.API.v1.addRoute('users.createToken', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('users.getPreferences', { authRequired: true }, {
+	get() {
+		const user = this.isUserFromParams() ? RocketChat.models.Users.findOneById(this.userId) : this.getUserFromParams();
```

Comment:
Using `this.getUserFromParams` you are allowing any user to get preferences of other users.